release/train_tests/benchmark/dataloader_factory.py

```diff
@@ -36,8 +521,268 @@ def get_ray_datasets(self) -> Dict[str, Dataset]:
         return {}
 
 
+class TorchDataLoaderFactory(BaseDataLoaderFactory, S3Reader):
```

Comment:
We can move this out to a `torch_dataloader_factory.py` and also the ray data stuff to a `ray_dataloader_factory.py`